src/Storages/ObjectStorage/DataLakes/DataLakeStorageSettings.h

```diff
@@ -63,6 +64,19 @@ Get latest metadata path from version-hint.text file.
     DECLARE(Int64, iceberg_format_version, 2, R"(
 Metadata format version.
 )", 0) \
+    DECLARE(DatabaseDataLakeCatalogType, iceberg_catalog_type, DatabaseDataLakeCatalogType::ICEBERG_REST, "Catalog type", 0) \
+    DECLARE(String, iceberg_catalog_credential, "", "", 0)             \
```

Comment:
Currently we allow storage settings to be specified as part of create database query and then apply it to storage: https://github.com/ClickHouse/ClickHouse/blob/b181da711bfad81474d3f5a35817f3ff0c8efa11/src/Databases/DataLake/DatabaseDataLake.cpp#L394-L395
But it means now we would be able to write `CREATE DATABASE glue_catalog .... SETTINGS catalog_credential = ..., iceberg_catalog_credential = ...`, which will be ambiguous. I guess the least we could do is to deduplicate them and throw an error if matching settings (with and without `iceberg_` prefix) are specified, or do you have a better idea?